Add feedback markers for plate being hit and being suppressed

Alien314 opened the pull request:
Adds a setting to scale the damage marker, and randomizes direction by up to a few degrees to prevent the marker from being exact.
Adds functionality to trigger the damage marker from the suppressed event, displaying in a different color when damage is 0.

Alien314 committed:
Scale Damage Marker setting and Suppression Marker

Alien314 committed:
Scale Damage Marker setting and Suppression Marker

Alien314 committed:
Settings and Suppressed Marker functionality

Alien314 committed:
Settings and Suppressed Marker functionality

Alien314 committed:
LIST paas instead of slider setting

Alien314 committed:
LIST paas instead of slider setting

Alien314 committed:
New paas

Alien314 committed:
New paas

diwako commented on addons/main/functions/fnc_showDamageFeedbackMarker.sqf:
```diff
@@ -11,14 +11,16 @@ private _selfOrUnkownDamage = isNull _instigator || {_unit isEqualTo _instigator
 private _ctrl = _display ctrlCreate ["RscPictureKeepAspect", -1];
 _ctrl ctrlSetBackgroundColor [0, 0, 0, 1];
 _ctrl ctrlSetPosition [0, 0, 1, 1];
```
TBH you do not need more files for the damage markers, you just need to change this and make that smaller or bigger. Then you can just use a slider CBA setting.

Alien314 replied:
I tried slider for a while but I wanted to maintain the distance from the screen center, and making the math work with that gap was beyond me.

diwako replied:
Fairly easy here. The control is just a box. it has its anchor point at 0,0. it is 1 by 1. That is the so called safe box in arma, it is a 4:3 box in the middle of the screen.

1 is just the width and height of that box

If you have a slider. lets say you want to make it half as large, you set the slider to 0.5. which translates the width and length to be `1 * 0.5 = 0.5`
Same for the position. Take the output of the size and divide that like so `(1 - 0.5) / 2 = 0.25`

Alien314 replied:
I got there, I just wanted to also push the marker away from center screen because lower scale would make it crowd the point of aim. I couldn't get a consistent distance and was doing stuff like `(((-1 * (_scale - 1)) ^ 3) * (1 - sin _angle) )`.

diwako replied:
that is just complicating things. ctrlSetAngle does not rotate the control itself, just the texture of the control within. 
if you want something that moves away from the center of the screen that does not involve just scaling the ctrl then you need to redo basically everything how it is handled right now.

Alien314 replied:
Yeah, that's why I gave up and did new images, wanted to have them recolorable anyway.
<img width="606" height="587" alt="image" src="https://github.com/user-attachments/assets/204c9ca4-d33a-44c0-84b7-07d480de1e0b" />

diwako replied:
frankly it should be another shake to make it even clearer that it is something different than a damage indicator.

For the whole moving away from center part, if you want i can take a look as well later this week.

diwako commented on addons/main/initSettings.inc.sqf:
```diff
@@ -178,6 +178,43 @@ _category = [_header, LLSTRING(subCategoryFeedback)];
+        GVAR(suppressedInit) = ["CAManBase", "Suppressed", { params ["_unit", "_distance", "_shooter"];
+            if (_unit isNotEqualTo (call CBA_fnc_currentUnit)) exitWith {};
+            if (_distance > ( (GVAR(suppressedMarker) min GVAR(suppressedMarkerLimit)) + 1) ) exitWith {};
+            [_unit,_shooter,0] call diw_armor_plates_main_fnc_showDamageFeedbackMarker;
```
i know it is not yet done, but remember you can use the `FUNC` macro here.

Also if possible do not add logic into the cba settings code body once it is final.

Alien314 replied:
Should I remove that, leave the need restart flag, and call it good?

diwako replied:
Imo best way to handle that is to put that EH into the post init, have a checkbox cba setting to enable or disable it.

Then in the EH itself before you do anything else, you just check if it is enabled, if it is not, exit out of the EH.

Alien314 committed:
getRelDir, inaccuracy, plate hit, ace feedback fix

Changed suppression indicator from 0 to -1, because plates call with 0 and made 0 trigger plate color instead.
Damage marker wasn't triggering on empty plates for ACE.

Alien314 committed:
getRelDir, inaccuracy, plate hit, ace feedback fix

Changed suppression indicator from 0 to -1, because plates call with 0 and made 0 trigger plate color instead.
Damage marker wasn't triggering on empty plates for ACE.

Alien314 committed:
Enable setting

Alien314 committed:
Enable setting

Alien314 committed:
Marker color settings

Alien314 committed:
Marker color settings

Alien314 committed:
suppression image

diwako committed:
fix hemtt lint

diwako committed:
slight optimizations and remove scaling for now (#168)
